src/main/java/org/jenkinsci/plugins/workflow/cps/InterpolatedUninstantiatedDescribable.java

```diff
@@ -0,0 +1,29 @@
+package org.jenkinsci.plugins.workflow.cps;
+
+import org.jenkinsci.plugins.structs.describable.UninstantiatedDescribable;
+
+import java.util.Map;
+import java.util.Set;
+
+public class InterpolatedUninstantiatedDescribable extends UninstantiatedDescribable {
```

Comment:
So after https://github.com/jenkinsci/workflow-cps-plugin/pull/370/files#r492091888 failed, I just extended `UninstantiatedDescribable` to hold a list of interpolated strings to see how that would work. If this is the way to go, it would seem better to either place this class in `structs` or modify `UninstantiatedDescribable` directly?

Reply:
Another possible way is a bit of a hack where I could store the set of interpolated strings in the `arguments` map of the `UninstantiatedDescribable` with some unique key.

Reply:
Hmm, yeah it's a bit awkward to have to inherit from `UninstantiatedDescribable` to get things working.

> If this is the way to go, it would seem better to either place this class in structs or modify UninstantiatedDescribable directly?

I'm not sure. The current use case is pretty specific to `workflow-cps`, but I guess we could extend `DescribableModel.instantiate` to have it accept information about sensitive variables so it could check for interpolation [here](https://github.com/jenkinsci/structs-plugin/blob/77a382758cceffdbd97c2d46f24f5d6ec8841c66/plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java#L454-L456), but that would be a pretty different approach.

> store the set of interpolated strings in the arguments map of the UninstantiatedDescribable with some unique key

Yeah, I'm not sure. Using a special key may be a little more robust since we wouldn't have to worry about changes to `UninstantiatedDescribable` breaking this plugin, but doesn't `structs` log a warning when there are unused parameters when it is instantiating something?

Reply:
passing interpolated data through `NamedArgsAndClosure` in 1afab594b3f0b0364a7bbed5aa6bbaf597bc19a0